app/api/v1/routes.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from sqlalchemy import or_, and_
+from flask import jsonify, make_response, Blueprint, request, abort
+
+from data.database import DEFAULT_DATABASE as db
+from data.models import Vulnerability, Nvd, Description, Cpe
+
+bp = Blueprint("api_v1", __name__, url_prefix="/api/v1")
+
+
+@bp.app_errorhandler(404)
+def api_404(e=None):
+    """Return a 404 in JSON format."""
+    if request.path.startswith('/api'):
+        return make_response(jsonify({'error': 'Not found', 'code': 404}), 404)
+    else:
+        return e
+
+
+@bp.app_errorhandler(500)
+def api_500(e=None):
+    """Return a 500 in JSON format."""
+    if request.path.startswith('/api'):
+        return make_response(
+            jsonify({
+                'error': 'Internal server error',
+                'code': 500
+            }), 500)
+    else:
+        return e
+
+
+@bp.route("/product/<vendor_id>/<product_id>")
+def vulns_by_product(vendor_id=None, product_id=None):
+    """View vulns associated to product."""
+    if product_id is None or product_id is None:
+        return abort(404)
+    nvd_ids = db.session.query(Cpe.nvd_json_id).filter(
+        and_(Cpe.vendor == vendor_id,
+             Cpe.product == product_id)).distinct().all()
+    count = len(nvd_ids)
+    cve = db.session.query(Nvd.cve_id).filter(Nvd.id.in_(nvd_ids)).all()
+    return jsonify({"count": count, "cve_ids": [x for x, in cve]})
+
+
+def _cpesToJson(products):
+    """Jsonify Cpes for API routes."""
+    count = len(products)
+    return jsonify({
+        "count":
+        count,
```

Comment:
Could you put this in the same line with the key?

Reply:
That would be great but unfortunatly `yapf` formatter sets it like that all the time. As I was unable to find any options, I simply specified yapf to ignore block like this.

```python
    return jsonify({
        "count": count,
        "products": [{
            'product': x,
            'vendor': y
        } for x, y, in products]
    })  # yapf: disable
```
Certainly not the best way but gets the job done